cluster-autoscaler/simulator/scheduler_based_predicates_checker.go

```diff
@@ -0,0 +1,339 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simulator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	kube_client "k8s.io/client-go/kubernetes"
+	v1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog"
+	scheduler_apis_config "k8s.io/kubernetes/pkg/scheduler/apis/config"
+	scheduler_plugins "k8s.io/kubernetes/pkg/scheduler/framework/plugins"
+	scheduler_framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	scheduler_listers "k8s.io/kubernetes/pkg/scheduler/listers"
+	scheduler_nodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+	scheduler_volumebinder "k8s.io/kubernetes/pkg/scheduler/volumebinder"
+	// We need to import provider to initialize default scheduler.
+	"k8s.io/kubernetes/pkg/scheduler/algorithmprovider"
+)
+
+// SchedulerBasedPredicateChecker checks whether all required predicates pass for given Pod and Node.
+// The verification is done by calling out to scheduler code.
+type SchedulerBasedPredicateChecker struct {
+	framework                scheduler_framework.Framework
+	delegatingSharedLister   *DelegatingSchedulerSharedLister
+	informerBasedShardLister scheduler_listers.SharedLister
```

Comment:
s/Shard/Shared

Reply:
This is not used anymore after you removed legacy implementations of predicate checking. Drop it?